src/libcrun/linux.c

```diff
@@ -1761,6 +1761,42 @@ libcrun_do_pivot_root (libcrun_container_t *container, bool no_pivot, const char
   return 0;
 }
 
+/* If one of stdin, stdout, stderr are pointing to /dev/null on
+ * the outside of the container, this moves it to /dev/null inside
+ * of the container. This needs to run afer pivot/chroot-ing. */
+int
+libcrun_reopen_dev_null(libcrun_error_t *err)
+{
+  struct stat dev_null;
+  struct stat statbuf;
+  int fd;
+  int i;
+
+  /* Open /dev/null inside of the container. */
+  fd = open ("/dev/null", O_RDWR);
+  if (fd == -1)
+    return crun_make_error (err, errno, "failed open()ing /dev/null");
+
+  if (fstat(fd, &dev_null) == -1) {
+    close(fd);
+    return crun_make_error (err, errno, "failed stat()ing /dev/null");
+  }
+
+  for (i = 0; i <= 2; i++) {
```

Comment:
another nit: please move the '{' on a new line and indent it (same below for the two function calls.